localenv/mock-account-servicing-entity/app/routes/accounts.create.tsx

```diff
@@ -12,16 +12,28 @@ import { createWallet } from '~/lib/wallet.server'
 import { messageStorage, setMessageAndRedirect } from '~/lib/message.server'
 import { createAccountSchema } from '~/lib/validate.server'
 import type { ZodFieldErrors } from '~/lib/types'
-import { getOpenPaymentsUrl } from '~/lib/utils'
+import { getOpenPaymentsUrl, getTenantCredentials } from '~/lib/utils'
 
 export async function loader() {
-  const assets = await loadAssets()
+  const session = await messageStorage.getSession()
+  const options = await getTenantCredentials(session)
+  const assets = await loadAssets(options)
 
-  return json({ assets })
+  const isTenant = process?.env?.IS_TENANT === 'true'
+
+  // Filter out assets that don't belong to the operator if the request returned all tenants' assets
+  return json({
+    assets: !isTenant
+      ? assets.filter(
+          (asset) => asset.node.tenantId === process?.env?.OPERATOR_TENANT_ID
+        )
+      : assets,
+    options
+  })
```

Comment:
not sure if im fully understanding this or not but would it make sense to do this server side? maybe as a new `filter` option in the `listAsset` ?

Reply:
Yeah honestly I think this is better off adding a filter that only the operator can use in the graphql query.

Reply:
Or just `tenantId` like we have for the other list queries (e.g. `outgoingPayments`)

Reply:
> Or just `tenantId` like we have for the other list queries (e.g. `outgoingPayments`)

I'll just add this now, so that the current filter doesn't get included when this gets merged.